packages/studio-plugin/src/ParsingOrchestrator.ts

```diff
@@ -262,12 +292,35 @@ export default class ParsingOrchestrator {
         `The pages directory does not exist, expected directory to be at "${this.paths.pages}".`
       );
     }
-    const files = fs.readdirSync(this.paths.pages, "utf-8");
-    return files.reduce((pageMap, filename) => {
-      const pageName = upath.basename(filename, ".tsx");
-      pageMap[pageName] = this.getPageFile(pageName);
-      return pageMap;
-    }, {} as Record<string, PageFile>);
+    return this.initNameToSourceFile(
+      this.paths.pages,
+      this.getOrCreatePageFile
+    );
+  }
+
+  private initLayoutNameToLayoutFile(): Record<string, LayoutFile> {
+    if (!fs.existsSync(this.paths.layouts)) {
+      return {};
+    }
+    return this.initNameToSourceFile(
+      this.paths.layouts,
+      this.getOrCreateLayoutFile
+    );
+  }
+
+  private initNameToSourceFile<SourceFile>(
```

Comment:
@oshi97, for this method, I think all we need is a more intuitive name.

Reply:
renamed to createFilenameMapping!